lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraMayaDataProducerSceneIndexData.cpp

```diff
@@ -53,8 +54,11 @@ MayaDataProducerSceneIndexData::MayaDataProducerSceneIndexData(FVP_NS_DEF::DataP
     : FVP_NS_DEF::DataProducerSceneIndexDataBase(params)
 {
     if (_dccNode) {
-        SetupUfeObservation(_dccNode);
+        SetupDCCNode();
     }
+    //Reset the prefix member which was used only for SetupDCCNode, as for usd stages we add a prefixing scene index outside of flow viewport which holds the prefix.
+    //This is a way to have a common call to MAYAHYDRA_NS::MhDataProducersMayaNodeToSdfPathRegistry::Get().Add in this class.
+    _prefix = SdfPath::AbsoluteRootPath();
```

Comment:
Could we not standardize the way we handle USD data to follow the same structure as other data producers? Looking quickly at the `HdPrefixingSceneIndex` creation for the USD data producer ([registration.cpp](https://github.com/Autodesk/maya-hydra/blob/827a42e6a11ded03e00e6a0e64d37371dc43e54e/lib/mayaHydra/hydraExtensions/sceneIndex/registration.cpp#L437-L441)), I don't see why we couldn't just rely on the prefixing scene index being added by the data producer API like [here](https://github.com/Autodesk/maya-hydra/blob/dev/827a42e6a11ded03e00e6a0e64d37371dc43e54e/flowViewport/API/perViewportSceneIndicesData/fvpDataProducerSceneIndexDataBase.cpp#L120-L131). This one is for non-USD data producers but I think we could do the same in the USD version of this method? And/or maybe use a `UsdImagingRootOverridesSceneIndex`?

Reply:
I recall I had tried that already but the prefixing scene index should be above all.